packages/ui-navigation/src/AppNav/index.tsx

```diff
@@ -187,48 +185,40 @@ class AppNav extends Component<AppNavProps> {
       </li>
     )
   }
-
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'items' implicitly has an 'any' type.
-  renderMenu(items) {
+  renderMenu(items: (ReactChild | ReactFragment | ReactPortal)[]) {
```

Comment:
The correct type here is this:

```suggestion
  renderMenu(items: React.ComponentElement<AppNavItemProps, Item>[]) {
```

We know that this component accepts only AppNav.Items as children, so these are always going to be AppNav.Item component elements. If you type it like this, you can remove the `const appNavItem = item as Item` casts from inside the method as well.

Reply:
I tried your suggestion but it caused an error since `renderMenu` will be called at line 259 
`{hiddenChildren.length > 0 && this.renderMenu(hiddenChildren)}` with the `hiddenChildren` is an array of `ReactChild | ReactFragment | ReactPortal` since it is the return value of `Children.toArray()`

Reply:
Yes, because the type needs to be corrected there as well. In the render method, when you set the `childrenArray`, it has to be typed to be an array of AppNav.Item children as well, because the component only accepts AppNav.Item children. The error will disappear after that.

```ts
const childrenArray = Children.toArray(children) as React.ComponentElement<AppNavItemProps, Item>[]
```